app-cpp/app355.cpp

```diff
@@ -184,6 +188,9 @@ void SatOrbit::SortOrbitalVector(std::vector<OrbitalData>& ioOrbitalVector)
     }
     else
     {
+        // lock the orbit mutex as both calculate and sort could be multithreaded
+        std::lock_guard<std::mutex> lock(mOrbitMutex);
```

Comment:
Same here... I don't see the data `mOrbitMutex` is meant to protect.
i see that a reference& to someone's vector is passed in as a parameter, but that doesn't necessarily have to be a `mOrbitalVector`.

any mutex un/locking should be co-located with the data it's supposed to protect.